.github/workflows/scripts/slack_notifications/gem_notifier.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'time'
+require 'httparty'
+require_relative 'slack_notifier'
+
+class GemNotifier < SlackNotifier
+  def self.check_for_updates(watched_gems)
+    return if gem_list_empty(watched_gems)
+    watched_gems.each do |gem_name|
+      verify_gem(gem_name) ? gem_info = verify_gem(gem_name) : return
+      versions = gem_versions(gem_info)
+      notifier.send_slack_message(gem_message(gem_name, versions)) if gem_updated?(versions)
+    end
+  end
+
+  def self.gem_list_empty(watched_gems)
+    if watched_gems.empty?
+      abort("Nothing to see here! The 'watched_gems' array cannot be empty")
+    end
+  end
+
+  def self.verify_gem(gem_name)
+    gem_info = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+
+    gem_info if gem_info.success?
```

Comment:
Currently an `HTTParty` GET failure will return `nil` which will result in an early `return`. There are 2 problems with this:

1. Elsewhere in this class, we use `abort(msg)` to explain a little bit about why we are stopping early
2. We have more info about what went wrong here in this method, and not back in the caller who hasn't been given anything more to go on than `nil`. The `gem_info` object could yield things like an HTTP status code.